src/machine.rs

```diff
@@ -1122,13 +1115,19 @@ impl<'mir, 'tcx> Machine<'mir, 'tcx> for MiriMachine<'mir, 'tcx> {
             None
         };
 
+        let def_id = frame.instance.def_id();
+        let is_local = (def_id.is_local() || ecx.machine.local_crates.contains(&def_id.krate))
+            && !frame.instance.def.requires_caller_location(ecx.machine.tcx);
```

Comment:
with this it isn't purely about `is_local` anymore. Maybe at least sprinkle some (doc) comments around to make it clear what the `is_local` is being used for

Reply:
The choice here hasn't been purely about whether the Instance is local for a while.

In fact, even the logic for `Machine::is_local(&FrameInfo)` which does only check if a Frame is local is only ever used in the diagnostics code, where we also check if a Frame is `#[track_caller]`. 

In any case, I've made an attempt at picking a less-wrong name. But I don't think it is good, and would appreciate any suggestion for a better one.

Reply:
I like the new name. :shrug: